bitbots_vision/bitbots_vision/vision.py

```diff
@@ -410,19 +375,18 @@ def _image_callback(self, image_msg):
         """
         # Drops old images and cleans up the queue.
         # Still accepts very old images, that are most likely from ROS bags.
-        image_age = rospy.get_rostime() - image_msg.header.stamp
-        if 1.0 < image_age.to_sec() < 1000.0:
-            rospy.logwarn(f"Vision: Dropped incoming Image-message, because its too old! ({image_age.to_sec()} sec)",
-                            logger_name="vision")
+        image_age = self.get_clock().now() - time.Time.from_msg(image_msg.header.stamp)
+        if 1.0 < image_age.nanoseconds / 1000000000 < 1000.0:
+            logger.warning(f"Vision: Dropped incoming Image-message, because its too old! ({image_age.to_sec()} sec)")
             return
 
         if self._transfer_image_msg_mutex.locked():
             return
 
         with self._transfer_image_msg_mutex:
-            # Transfer the image to the main thread
-            self._transfer_image_msg = image_msg
+            self._handle_image(image_msg)
 
+    @profile
```

Comment:
What happens if the import for profile failed, would this crash the vision?
(because profile is in a try/except block)